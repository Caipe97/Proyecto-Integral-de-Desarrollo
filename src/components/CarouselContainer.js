import {React, } from 'react';
//import Carousel from 'react-bootstrap/Carousel';
import MealModal from './MealModal';
import './CarouselContainer.css';
import 'bootstrap/dist/css/bootstrap.min.css';
import Carousel from "react-multi-carousel";
import "react-multi-carousel/lib/styles.css";
const responsive = {
  superLargeDesktop: {
    // the naming can be any, depends on you.
    breakpoint: { max: 4000, min: 3000 },
    items: 6
  },
  desktop: {
    breakpoint: { max: 3000, min: 1024 },
    items: 4
  },
  tablet: {
    breakpoint: { max: 1024, min: 464 },
    items: 3
  },
  mobile: {
    breakpoint: { max: 464, min: 0 },
    items: 2
  }
};

   

const CarouselContainer = (props) => {
    return (
        <Carousel responsive={responsive}>
        {props.meals.map((meal, index) =>
            <div className="card-container" key={index}>
                <div style={{ background: '#CCDDE2', textAlign: 'center', margin: 5, borderRadius: 10, paddingBottom: 5}}>
                    <div className="img" style={{fontSize: 24}}>
                        <p style={{textOverflow: 'ellipsis', overflow: 'hidden', whiteSpace: 'nowrap'}}>{meal.name}</p>
                    </div>
                    <div>
                        {meal.dateEaten.toString().replace('T', ' ').substring(0, 16)}
                    </div>
                    <div>
                        <MealModal meal={meal} history={props.history} onDeleteMeal={props.onDeleteMeal} onUpdateCurrentMealInState={props.onUpdateCurrentMealInState} />
                    </div>
                </div>
            </div>
         )}       
    </Carousel>
    )
}

export default CarouselContainer;