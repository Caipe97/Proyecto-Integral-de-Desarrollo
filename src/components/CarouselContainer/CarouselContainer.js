import {React, } from 'react';
//import Carousel from 'react-bootstrap/Carousel';
import GoalModal from '../Modal/GoalModal';
import './CarouselContainer.css';
import 'bootstrap/dist/css/bootstrap.min.css';
import Carousel from "react-multi-carousel";
import "react-multi-carousel/lib/styles.css";
import { Button } from 'react-bootstrap';
import { Link } from "react-router-dom";
const responsive = {
  superLargeDesktop: {
    // the naming can be any, depends on you.
    breakpoint: { max: 4000, min: 3000 },
    items: 6
  },
  desktop: {
    breakpoint: { max: 3000, min: 1024 },
    items: 4
  },
  tablet: {
    breakpoint: { max: 1024, min: 464 },
    items: 3
  },
  mobile: {
    breakpoint: { max: 464, min: 0 },
    items: 2
  }
};

   

const CarouselContainer = (props) => {
    return (
        <Carousel responsive={responsive}>
          
          <div className="card-container" key={'addMeta'}>
              <div style={{ background: '#CCDDE2', textAlign: 'center', margin: 5, borderRadius: 10, paddingBottom: 5}}>
              <Button  className="card-container" type="button" variant="primary" style={{marginBottom: 14, height: 50, width: "100%",backgroundColor: 'rgb(18, 207, 90)', borderColor: 'rgb(18, 207, 90)'}}>
              <Link to="/goals">Nueva Meta</Link>
              </Button>
              <Button  className="card-container" type="button" variant="primary" style={{height: 50, width: "100%",backgroundColor: 'rgb(18, 207, 90)', borderColor: 'rgb(18, 207, 90)'}}>
              Historial de Metas
              </Button>
              </div>
          </div>
        {props.goals.map((goal, index) =>
            <div className="card-container" key={index}>
                <div style={{ background: '#CCDDE2', textAlign: 'center', margin: 5, borderRadius: 10, paddingBottom: 5}}>
                    <div className="img" style={{fontSize: 24}}>
                        <p style={{textOverflow: 'ellipsis', overflow: 'hidden', whiteSpace: 'nowrap'}}>{goal.name}</p>
                    </div>
                    <div>
                        {goal.dateStart.toString().replace('T', ' ').substring(0, 7)}
                    </div>
                    {goal 
                    ?                     
                    <div>
                      <GoalModal goal={goal} history={props.history} onDeleteGoal={props.onDeleteGoal} onUpdateCurrentGoalInState={props.onUpdateCurrentGoalInState} />
                    </div>
                    : null
                    }

                </div>
            </div>
         )}       
    </Carousel>
    )
}

export default CarouselContainer;