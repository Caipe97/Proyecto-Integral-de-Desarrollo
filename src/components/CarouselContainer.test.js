import React from 'react';
import { shallow } from 'enzyme'; 
import CarouselContainer from './CarouselContainer';
//<CarouselContainer meals={props.meals} history={history}/>
describe('Carousel tests', () => {
    //userId:1
    const mockPropsMeal=
    [{
        mealId: 21,
        name: "queso",
        foodsAndQuantity: {quantity: 1, foods: {foodId: 1, name: 'Milanesa', recommendedServing: 85, caloriesPerServing: 198, createdAt: '2021-09-15T19:58:04.486Z'}},
        dateEaten: "2021-09-09T00:00:00.000Z",
        userId: 1
    }];
    const historyMock = { push: jest.fn() };
    //const mockPropsMeal = {meals:meal1 };     
    let wrapper;

    beforeEach(() => {
        wrapper = shallow(<CarouselContainer meals={mockPropsMeal} history={historyMock}/>);
    })

    it('expect to render carousel', () => {
        expect(wrapper).toMatchSnapshot();
        //   expect((wrapper.props.meals).map(obj => obj.mealId)).toEqual(
        //     mockPropsMeal.map(obj => obj.mealId)
        //   );
        expect(wrapper.find('Carousel.Item')).toBeDefined();
        expect(wrapper.find('.Carousel.Item.div.div.div.p').contains('Milanesa')).toBeDefined();
        //expect(wrapper2.find('Carousel.Item').contains('comida1')).toBeDefined(); 
        //expect(wrapper.find('Dropdown').to.have.length(2));
    })
    // it('expect to render Header component to render with meals for user', () => {
    //     //expect(wrapper).toMatchSnapshot();
    //     expect(wrapper.find('.modal-footer').length).toEqual(1);  
    // })

})
