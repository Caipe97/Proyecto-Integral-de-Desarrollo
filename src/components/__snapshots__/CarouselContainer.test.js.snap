// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Carousel tests expect to render carousel 1`] = `
<Carousel
  controls={true}
  defaultActiveIndex={0}
  fade={true}
  indicators={true}
  interval={5000}
  keyboard={true}
  nextIcon={
    <span
      aria-hidden="true"
      className="carousel-control-next-icon"
    />
  }
  nextLabel="Next"
  pause={false}
  prevIcon={
    <span
      aria-hidden="true"
      className="carousel-control-prev-icon"
    />
  }
  prevLabel="Previous"
  slide={true}
  touch={true}
  wrap={true}
>
  <CarouselItem
    key="21"
  >
    <div
      className="container"
      style={
        Object {
          "background": "lightblue",
          "borderRadius": "10px",
          "textAlign": "center",
        }
      }
    >
      <div
        className="row"
      >
        <div
          className="col-md-4 col-lg-12"
        >
          <p>
            queso
          </p>
          <p>
            2021-09-09 00:00
          </p>
        </div>
        <div
          className="col-md-4 col-lg-12"
          style={
            Object {
              "margin": "auto",
              "position": "center",
              "textAlign": "center",
              "width": "30%",
            }
          }
        >
          <p>
            Calorias Consumidas
          </p>
        </div>
        <div
          className="col-md-4 col-lg-12"
          style={
            Object {
              "marginBottom": "10px",
              "marginTop": "10px",
              "position": "center",
            }
          }
        >
          <MealModal
            history={
              Object {
                "push": [MockFunction],
              }
            }
            meal={
              Object {
                "dateEaten": "2021-09-09T00:00:00.000Z",
                "foodsAndQuantity": Object {
                  "foods": Object {
                    "caloriesPerServing": 198,
                    "createdAt": "2021-09-15T19:58:04.486Z",
                    "foodId": 1,
                    "name": "Milanesa",
                    "recommendedServing": 85,
                  },
                  "quantity": 1,
                },
                "mealId": 21,
                "name": "queso",
                "userId": 1,
              }
            }
          />
        </div>
      </div>
    </div>
  </CarouselItem>
</Carousel>
`;
